src/nbd.c

```diff
@@ -860,6 +861,10 @@ gboolean r_nbd_run_server(gint sock, GError **error)
 			code = curl_easy_getinfo(msg->easy_handle, CURLINFO_PRIVATE, &xfer);
 			g_assert(code == CURLE_OK);
 
+			code = curl_easy_getinfo(xfer->easy, CURLINFO_RESPONSE_CODE, &response_code);
```

Comment:
Can't we use
```suggestion
			code = curl_easy_getinfo(msg->easy_handle, CURLINFO_RESPONSE_CODE, &response_code);
```
here?